### ISSUE TITLE
Problem with mentions' autocomplete

### ISSUE THREAD

**Issue by remouk** [Automattic/wordpress-activitypub#272]
Hello, I'm not really sure how this is working, but I think this is related to #213 where the autocomplete is messing up content.

I have a page where I give my contacts: where to find me / follow me. For Mastodon/ActivityPub I wrote "@myusername@mywebsite.com" in plain text.

Since the update, it's automatically replaced with "@myusername" which is not plain text but a link to "https://mywebsite.com/author/myusername"

I cannot bypass the change, thus I cannot simply share my ActivityPub account anymore.

Am I missing something?

**Comment by pfefferle** [comment]
yes, that is currently a bit tricky. You can add some hacks with adding HTML like for example:

`<span>@myusername</span>@mywebsite.com`

**Comment by remouk** [comment]
Thanks for the workaround! 👍

Do you agree this should not happen? I didn't use the autocomplete and never chose this behavior, it should not be automatically applied, right?

**Comment by pfefferle** [comment]
It should automatically apply, this is how the feature is planned. I have no idea to do it differently. It now works exactly how it works on mastodon. And mastodon has the same issue. You will not be able to post the plain identifier.

I think this is the default behavior, but if you can tell me your exact usecase, I maybe find a different and more elegant way to share the WebFinger URI in a blogpost.

**Comment by remouk** [comment]
Maybe this is just me, sorry, but I don't think you should apply the feature exactly like Mastodon does. WordPress has it's own way of working, and users expects the software to work that way.

In WordPress, there shouldn't be a case where the text is modified without our control. You write and arrange your posts in blocks (or classic editor), this is WYSIWYG. I write text in the editor = it is shown as text = it should display as text.

For example, if you type an url or an email address in WP it will be simple text. It's up to you to make an hyperlink. While in Mastodon an hyperlink is automatically made clickable, in WordPress it is not.

Autocomplete is a good idea but it should be only be applied if I clicked on it. Right now, it's out of our control. Worse, it applies automatically to old posts as well!

It took me a while to understand what was happening, and I understood because I saw the problem the same day as I made the update. But I guess many users won't understand why their content is modified.

**Comment by mediaformat** [comment]
> I have a page where I give my contacts: where to find me / follow me. For Mastodon/ActivityPub I wrote "@myusername@mywebsite.com" in plain text.

As @pfefferle mentioned, being able to `@` mention people is part of Wordpress core, and the plugin (despite the current lack of UI for remote users), so your issue, however you look at it is a question of escaping current functionality to achieve a desired presentation. 
Here are some suggestions 
- write it as plain text and link it yourself 
- or wrap it in another tag, ex: `<pre>`

**Comment by remouk** [comment]
"_write it as plain text and link it yourself_"
-> That's what I'm asking for. I'd love to write it in plain text, but I can't. :/

"_or wrap it in another tag, ex: `<pre>`_"
-> That's what I did in the first place, but let's be honest this is an ugly workaround.

I'm sorry to insist, but mentions should only be applied if we want to. Modifying content automatically, and having no way to bypass the change, is not something a WordPress user expects. This is a really weird and unwanted behavior.

The autocomplete is a good idea, but if we don't use it, if we don't click it, it should not modify the plain text we just wrote. If we click it, there should be a way to remove the mention, too. Just like an hyperlink works, for example.

Btw, thank you very much for your hard work, all your plugins are very helpful and very well done!
